pkg/controller/pxc/users.go

```diff
@@ -171,27 +172,26 @@ func (r *ReconcilePerconaXtraDBCluster) handleRootUser(cr *api.PerconaXtraDBClus
 
 	logger.Info(fmt.Sprintf("User %s: password changed, updating user", user.Name))
 
-	// update pass
 	err := r.updateUserPass(cr, secrets, internalSecrets, user)
 	if err != nil {
 		return errors.Wrap(err, "update root users pass")
 	}
 	logger.Info(fmt.Sprintf("User %s: password updated", user.Name))
 
-	// syncProxyUser
-	err = r.syncPXCUsersWithProxySQL(cr)
-	if err != nil {
-		return errors.Wrap(err, "sync users")
+	if cr.Spec.ProxySQL != nil && cr.Spec.ProxySQL.Enabled {
+		err = r.syncPXCUsersWithProxySQL(cr)
```

Comment:
this function is no-op if proxysql is not enabled. maybe we just need to move the log inside the function